compiler/rustc_privacy/src/lib.rs

```diff
@@ -251,6 +251,12 @@ where
                     })?;
                 }
             }
+            ty::TyAlias(def_id, _) => {
```

Comment:
to support existing privacy rules, maybe we just want to `.type_of().subst()` and recurse on that?

Reply:
Shouldn't we check that the alias itself is public? Or is it what you're saying and that I should instead check it by using the `subst`?

Reply:
Do we _want_ to check that the alias is public? That's a language change, is it not? Since it affects code that currently compiles.

Reply:
I can't find a case where it would change the behaviour. Do you have one in mind by any chance? Like that we could even add it as a ui test.

Reply:
Hmm.. you may be right. I think it's okay because we unconditionally walk the substs below.